### PR TITLE
Better support for git submodule

### PR DESCRIPTION
**Pull request by cristiklein**
This fix make cpplint correctly detect the `root` of a source file located in a git submodule.

### CODE REVIEW

File: cpplint.py [hunk 1]
```diff
@@ -1318,8 +1318,10 @@ def RepositoryName(self):
       # searching up from the current path.
       root_dir = current_dir = os.path.dirname(fullname)
       while current_dir != os.path.dirname(current_dir):
-        if (os.path.exists(os.path.join(current_dir, ".git")) or
-            os.path.exists(os.path.join(current_dir, ".hg")) or
+        if (os.path.exists(os.path.join(current_dir, ".git"))):
+          root_dir = current_dir
+          break  # stop here, we might be in a submodule
+        if (os.path.exists(os.path.join(current_dir, ".hg")) or
```
Comment (tkruse) [inline cpplint.py]: the same argument would apply to Mercurial subrepositories, no?

Comment (cristiklein) [reply cpplint.py]: I haven't used Mercurial for years, so I would rather someone knowledgeable could check.

Comment (tkruse) [reply cpplint.py]: Not a reason not to merge anyway.

File: cpplint.py [hunk 2]
```diff
@@ -1318,8 +1318,10 @@ def RepositoryName(self):
       # searching up from the current path.
       root_dir = current_dir = os.path.dirname(fullname)
       while current_dir != os.path.dirname(current_dir):
-        if (os.path.exists(os.path.join(current_dir, ".git")) or
-            os.path.exists(os.path.join(current_dir, ".hg")) or
+        if (os.path.exists(os.path.join(current_dir, ".git"))):
```
Comment (tkruse) [inline cpplint.py]: pylint does not like the outer parentheses in this case, hence the travis failure.

**Pull request closed without merging**